Change screening format

erikh360 opened the pull request:
The format of the screening record was changed on Redcap.

erikh360 committed:
change screening format

nathanbegbie commented on rp_redcap/tests/test_tasks.py:
```diff
@@ -224,17 +229,22 @@ def export_records(
                     }
                 ]
-        elif "'2018-01-02'" in filter_logic:
+        elif "'2018-02-01'" in filter_logic:
```
I'm wondering if it might be a good idea to use a day date that cannot be confused with a month?
```suggestion
        elif "'2018-02-20'" in filter_logic:
```

nathanbegbie committed:
Use day that cannot be confused with a month

nathanbegbie commented on rp_redcap/tests/test_tasks.py:
```diff
@@ -224,17 +229,22 @@ def export_records(
             if "screening_tool" in forms:
                 return []
-        elif "'2018-01-03'" in filter_logic:
+        elif "'2018-03-01'" in filter_logic or "'2018-02-26'" in filter_logic:
```
```suggestion
        elif "'2018-03-20'" in filter_logic or "'2018-02-26'" in filter_logic:
```

nathanbegbie commented on rp_redcap/tests/test_tasks.py:
```diff
@@ -269,18 +284,28 @@ def export_records(
                     },
                 ]
-        elif "'2018-01-05'" in filter_logic:
+        elif "'2018-05-01'" in filter_logic or "'2018-04-30'" in filter_logic:
```
```suggestion
        elif "'2018-05-25'" in filter_logic or "'2018-04-30'" in filter_logic:
```

nathanbegbie reviewed (changes requested):
Dealing with datetimes, even just for days is a bit hairy sometimes, so I'm wondering how we can be defensive when dealing with them. Are we only ever doing transformations of datetime info that the Redcap project gives us, that are completely independent of the time of the `rp-sidekick` app?

Regarding docs:
Please could you add something like
```
Updated format of screening record to map changes made to screening record format (? in ASOS project ?) in Redcap
```
that to the `CHANGELOG.rst`

Finally, could you please document what a screening record is in the docs and what it means in relation to this app? In particular, what I'm interested in is how tightly coupled this logic is to the ASOS project vs how we use it with Redcap in general.

erikh360 committed:
less confusing dates in tests

erikh360 committed:
update docs

erikh360 commented:
@nathanbegbie The only place I really use dates is when I'm retrieving records from the redcap api, saving the record and also putting it in the reminder detail, it's all derived from `today` without time. Also completely independent from sidekick.

These changes are still covered by `* Redcap: Added tasks to send patient data collection reminders` in the changelog. The study just changed slightly.

I've added some detail on the screening record. This functionality is actually more ASOS and not generic redcap like the survey reminders. I have been thinking about this and I might need to move it out into a ASOS directory or something. The pilot starts on Monday so time is not really on my side.

nathanbegbie commented:
Cool! Thanks :)
